core/client/fs/src/main/java/alluxio/client/block/stream/GrpcBlockingStream.java

```diff
@@ -215,12 +229,20 @@ private AlluxioStatusException toAlluxioStatusException(Throwable t) {
       if (ex.getStatus() == Status.CANCELED) {
         // Streams are canceled when server is shutdown. Convert it to UnavailableException for
         // client to retry.
-        ex = new UnavailableException("Stream is canceled by server.", ex);
+        ex = new UnavailableException(formatErrorMessage("Stream is canceled by server."), ex);
       }
     } else {
       ex = AlluxioStatusException.fromThrowable(mError);
     }
-    return ex;
+    // attaches description to the exception while maintaining the cause
+    return (AlluxioStatusException) AlluxioStatusException
+        .from(ex.getStatus(), formatErrorMessage(ex.getMessage()))
+        .initCause(ex.getCause());
+  }
+
+  private String formatErrorMessage(String format, Object... args) {
+    StringBuilder errorMessage = new StringBuilder(String.format(format, args));
+    return new StringBuilder(errorMessage).append(String.format(" (%s)", args)).toString();
```

Comment:
shouldn't `mDescription` be in the message somewhere?

Reply:
That's a typo :( fixed.